Lib/logging/__init__.py

```diff
@@ -997,6 +997,26 @@ def emit(self, record):
         except Exception:
             self.handleError(record)
 
+    def setStream(self, stream):
+        """
+        Sets the StreamHandler's stream to the specified value,
+        if it is different.
+
+        Returns the old stream, if the stream was changed, or None
+        if it wasn't.
+        """
+        if stream is self.stream:
+            result = None
+        else:
+            result = self.stream
+            self.acquire()
+            try:
+                self.flush()
```

Comment:
This may seem a bit defensive but is unlikely to be in performance-critical code, I would have thought.